blob/store.go

```diff
@@ -24,11 +24,28 @@ import (
 	"sort"
 )
 
-// A KV represents a mutable key-value store in which each value is identified
-// by a unique, opaque string key.  An implementation of KV is permitted (but
-// not required) to report an error from Put when given an empty key.  If the
-// implementation cannot store empty keys, it must report ErrKeyNotFound when
-// operating on an empty key.
+// A Store represents a collection of key-value namespaces ("keyspaces")
+// identified by string labels. Each keyspace in a store is logically distinct;
+// the keys from one space are independent of the keys in another.
+//
+// Implementations of this interface must be safe for concurrent use by
+// multiple goroutines.
+type Store interface {
+	// Keyspace returns a key space on the store.
+	Keyspace(name string) (KV, error)
+
+	// Close allows the store to release any resources held open while in use.
+	// If an implementation has nothing to release, it must report nil.
+	// After closing a Store, any keyspaces derived from the store should report
+	// errors when used.
+	Close(context.Context) error
```

Comment:
Doc question: is Close() required for correctness? if my code decides to ignore Store.Close() and just drop the Store into the GC, do I stand to lose data?

Reply:
It's mainly a leak of the implantation. Some storage engines lose state if you don't close.

Reply:
Per our later discussion, I have removed `Close` from this and the `KV` interface, and reintroduced the (previously removed) `blob.Close` helper that closes if the concrete type has a corresponding method, or otherwise is a no-op.